### ISSUE TITLE
enforce same shape for input / output in multinomial case

### ISSUE THREAD

**Issue by jasmainak** [pavanramkumar/pyglmnet#66]
This issue was discovered here: https://github.com/pavanramkumar/pyglmnet/commit/749bba9559bcddc65b0c5f68b7196691a5b5f489#commitcomment-17402965

basically if the output `y_pred` is going to be `n_samples x n_classes`, the input `y` should also be the same shape. Converting to this shape can be done by

1. the user 
2. using scikit-learn 
3. a convenience function for this can be provided.

but not inside `GLM` class.

**Comment by pavanramkumar** [comment]
@jasmainak I'm revisiting this issue now. I feel that we should minimize the burden on the user. 
If we would force `y` to be `n_samples x n_classes`, we either need to provide a helper function which leads to overhead in terms of API, or we need to make then use `scikit-learn`.

Since it's much more standard for many problems to have labels as integers in a 1-d array (even sklearn's `make_classification` does it this way), I feel that we can afford to leave the API as is.

If you agree, I can close this issue!

**Comment by jasmainak** [comment]
@pavanramkumar do you agree that the shape of the input in fit() should be
the same as the shape of the output in transform()? If we agree to make it
1-D we have to still internally do the transformation back to 1-D before
returning to the user. If we allow 1-D, imagine the range of possible ways
the label could be encoded. You could have two strings for 0 and 1 or two
numbers etc. We would have to take care of all these different
possibilities. Or use sklearn LabelEncoder internally ...

**Comment by pavanramkumar** [comment]
getting back to this again @jasmainak 

seems there's no easy way other than to make a separate method called `predict_proba()` just for `multinomial`.

**Comment by jasmainak** [comment]
@pavanramkumar the `predict_proba` method would predict probabilities instead of class labels, no?
